htdocs/product/card.php

```diff
@@ -1434,8 +1434,8 @@
                 print '<tr><td class="nowrap">';
                 print '<table width="100%" class="nobordernopadding"><tr><td class="nowrap">';
                 print $langs->trans("BarcodeType");
-                print '</td>';
-                if (($action != 'editbarcodetype') && ! empty($user->rights->barcode->creer)) print '<td align="right"><a href="'.$_SERVER["PHP_SELF"].'?action=editbarcodetype&amp;id='.$object->id.'">'.img_edit($langs->trans('Edit'),1).'</a></td>';
+                print '<td>';
+                if (($action != 'editbarcodetype') && ! empty($user->rights->barcode->creer_advance)) print '<td align="right"><a href="'.$_SERVER["PHP_SELF"].'?action=editbarcodetype&amp;id='.$object->id.'">'.img_edit($langs->trans('Edit'),1).'</a></td>';
```

Comment:
It should be 
if (($action != 'editbarcodetype') && ! empty($user->rights->product->creer))
We can edit property of product if we have permission to edit product.
And then we must introduce test on $user->rights->barcode->creer_advance but only if constant that say advanced perm are enabled.